test/e2e/end_user_test.go

```diff
@@ -51,4 +73,94 @@ var _ = Describe("Openshift on Azure end user e2e tests [EndUser]", func() {
 		// Expect(kerrors.IsForbidden(err)).To(Equal(true))
 		fmt.Printf("PDB create error: %v\n", err)
 	})
+
+	It("should deploy a template and check the visit counter increments", func() {
+		const NAME = "nodejs-mongodb-example"
+		var regex = regexp.MustCompile(`\<span.*id=\"count-value\"\>(?P<counter>\d+)\<\/span\>`)
+
+		// Create the template
+		template, err := c.tc.Templates("openshift").Get(
+			NAME, metav1.GetOptions{})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Create a secret to hold the template params
+		secret, err := c.cc.Secrets(c.namespace).Create(&corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "parameters",
+			},
+			StringData: map[string]string{
+				"NAME":         NAME,
+				"MEMORY_LIMIT": "1024Mi",
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Instantiate the template
+		ti, err := c.tc.TemplateInstances(c.namespace).Create(
+			&templatev1.TemplateInstance{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "templateinstance",
+				},
+				Spec: templatev1.TemplateInstanceSpec{
+					Template: *template,
+					Secret: &corev1.LocalObjectReference{
+						Name: secret.Name,
+					},
+				},
+			})
+		Expect(err).NotTo(HaveOccurred())
+
+		// Wait for the deployments to finish
+		watcher, err := c.tc.TemplateInstances(c.namespace).Watch(
```

Comment:
Instead of L114-L144 you can simply do
```go
err := wait.PollImmediate(2*time.Second, time.Minute, t.templateInstanceIsReady)
Expect(err).NotTo(HaveOccurred())
```
and inside `conditions_test.go`
```go
func (t *testClient) templateInstanceIsReady() (bool, error) {
	ti, err := t.tc.TemplateInstances().Get(t.namespace, metav1.GetOptions{})
	if kerrors.IsNotFound(err) {
		return false, nil
	}
	if err != nil {
		return false, err
	}

	for _, cond := range ti.Status.Conditions {
		if cond.Type == templatev1.TemplateInstanceReady &&
			cond.Status == corev1.ConditionTrue {
			return true, nil
		} else if cond.Type == templatev1.TemplateInstanceInstantiateFailure &&
			cond.Status == corev1.ConditionTrue {
			return false, fmt.Errorf("templateinstance %q failed", t.namespace)
		}
	}

	return false, nil
}
```
You will need to use c.namespace as the name of the template instance in that case. I think it's fine to do it for now.